utils/src/main/scala/za/co/absa/enceladus/utils/types/Format.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright 2018-${today.year} ABSA Group Limited
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.utils.types
+
+import java.security.InvalidParameterException
+import org.apache.spark.sql.types.{DataType, StructField}
+import scala.util.Try
+
+/**
+  * Class to carry enhanced information about formatting patterns in ocnversion from/to string
+  * @param pattern  the actual pattern to format the type conversion; if none global default pattern for the type is used
+  * @param forType  the type the format is intended for
+  */
+class Format(val pattern: Option[String], val forType: Option[DataType] = None){
+  private val actualFormat:String = pattern.getOrElse(Defaults.getGlobalFormat(forType.get))
+  def isDefault: Boolean = pattern.isEmpty
+  def get: String = actualFormat
+  def getOrElse(default: String): String =  pattern.getOrElse(default)
+
+  def isEpoch: Boolean = {
```

Comment:
Is there a reason to have `getOrElse` on one line but `isEpoch` and `epochFactor` as multiline?

Reply:
It's probably my perception of the class, what it models. And that the first is a private `val`, that is only initialized upon class creation; while the other two public are "functions". whose result can theoretically be dynamic.